### ISSUE TITLE
Inconsistent `unused_qualifications` lint

### ISSUE THREAD

**Issue by c410-f3r** [rust-lang/rust#99971]
For `alloc` imports, `unused_qualifications` shouldn't be triggered.

```rust
#![forbid(unused_qualifications)]
extern crate alloc;
type _Foo = alloc::string::String;  // Triggers!
```

As a reference, this "does-not-trigger" behaviour already happens with `core` imports.

```rust
#![forbid(unused_qualifications)]
type _Foo = core::task::Poll<i32>; // Does not trigger!
```

The lint becomes a problem when you are dealing with optional `no_std` code and imports `alloc` regardless of a `std` feature.

**Comment by c410-f3r** [comment]
My bad, `type _Foo = core::task::Poll<i32>` does not trigger because it isn't in the prelude.

I guess the lint is then behaving like it should, sorry.

**Comment by wcampbell0x2a** [comment]
> My bad, `type _Foo = core::task::Poll<i32>` does not trigger because `Poll` isn't in the prelude.
> 
> I guess the lint is then behaving like it should, sorry.

no problem, got me to read some rustc code

**Comment by c410-f3r** [comment]
Thank you @wcampbell0x2a !
